dra-evolution/pkg/api/capacity_types.go

```diff
@@ -87,7 +108,13 @@ type Device struct {
 	// +optional
 	Attributes []DeviceAttribute `json:"attributes,omitempty" protobuf:"bytes,3,opt,name=attributes"`
 
-	// TODO for 1.31: define how to support partitionable devices
+	// SharedAllocatableConsumed contains the pooled allocatable resources
+	// that are consumed when this device is allocated.
+	//
+	// Must not have more than 32 entries.
+	//
+	// +optional
+	SharedAllocatableConsumed map[string]resource.Quantity `json:"sharedAllocatableConsumed,omitempty"`
```

Comment:
Do we need a level of indirection here instead of just a map of `Quantity`? Otherwise we can never use anything other than a quantity in `ResourceCapacity`.

Reply:
probably, see the PR comment notes: https://github.com/kubernetes-sigs/wg-device-management/pull/27#issue-2339184965

Reply:
Would love your opinion specifically on the last question - pool-per-gpu or something else

Reply:
I personally like the idea of grouping them with names. It means I'm not *forced* to have one GPU per ResourceSlice if I don't want to. This will likely be important for "simpler" devices that only partition on, say, one dimention.

For example with this, I can have the following in a single ResourceSlice (using the newly proposed name for this field):
```
apiVersion: resource.k8s.io/v1alpha3
kind: ResourceSlice
metadata:
  name: <uuid>
spec:
  driver: gpu.nvidia.com
  poolName: gpus
  nodeName: node0
  sharedCapacity:
  - name: gpu-0-shared-capacity
    resources: 
    - name: memory
      capacity: 40Gi
    - name: compute
      capacity: 180
  - name: gpu-1-shared-capacity
    resources: 
    - name: memory
      capacity: 40Gi
    - name: compute
      capacity: 180
  devices:
  - name: gpu-0
    sharedCapacityConsumed:
    - name: gpu-0-shared-capacity
      resources: 
      - name: memory
        capacity: 40Gi
      - name: compute
        capacity: 180
  - name: gpu-0-first-half
    sharedCapacityConsumed:
    - name: gpu-0-shared-capacity
      resources: 
      - name: memory
        capacity: 20Gi
      - name: compute
        capacity: 90
  - name: gpu-0-second-half
    sharedCapacityConsumed:
    - name: gpu-0-shared-capacity
      resources: 
      - name: memory
        capacity: 20Gi
      - name: compute
        capacity: 90
  - name: gpu-1
    sharedCapacityConsumed:
    - name: gpu-1-shared-capacity
      resources: 
      - name: memory
        capacity: 40Gi
      - name: compute
        capacity: 180
  - name: gpu-1-first-half
    sharedCapacityConsumed:
    - name: gpu-1-shared-capacity
      resources: 
      - name: memory
        capacity: 20Gi
      - name: compute
        capacity: 90
  - name: gpu-1-second-half
    sharedCapacityConsumed:
    - name: gpu-1-shared-capacity
      resources: 
      - name: memory
        capacity: 20Gi
      - name: compute
        capacity: 90
```

Reply:
Agreed, I like this too.

Reply:
@klueska I pushed an update with this, PTAL.

Reply:
Patrick convinced me to go back to a flat list, which can accomplish the same with:
```
apiVersion: resource.k8s.io/v1alpha3
kind: ResourceSlice
metadata:
  name: <uuid>
spec:
  driver: gpu.nvidia.com
  poolName: node0
  nodeName: node0
  sharedCapacity:
  - name: gpu-0-shared-memory
    capacity: 40Gi
  - name: gpu-0-shared-compute
    capacity: 180
  - name: gpu-1-shared-memory
    capacity: 40Gi
  - name: gpu-1-shared-compute
    capacity: 180
  devices:
  - name: gpu-0
    sharedCapacityConsumed:
    - name: gpu-0-shared-memory
      capacity: 40Gi
    - name: gpu-0-shared-compute
      capacity: 180
  - name: gpu-0-first-half
    sharedCapacityConsumed:
    - name: gpu-0-shared-memory
      capacity: 20Gi
    - name: gpu-0-shared-compute
      capacity: 90
  - name: gpu-0-second-half
    sharedCapacityConsumed:
    - name: gpu-0-shared-memory
      capacity: 20Gi
    - name: gpu-0-shared-compute
      capacity: 90
  - name: gpu-1
    sharedCapacityConsumed:
    - name: gpu-1-shared-memory
      capacity: 40Gi
    - name: gpu-1-shared-compute
      capacity: 180
  - name: gpu-1-first-half
    sharedCapacityConsumed:
    - name: gpu-1-shared-memory
      capacity: 20Gi
    - name: gpu-1-shared-compute
      capacity: 90
  - name: gpu-1-second-half
    sharedCapacityConsumed:
    - name: gpu-1-shared-memory
      capacity: 20Gi
    - name: gpu-1-shared-compute
      capacity: 90
```
I will add it in the KEP as such.